### DISCUSSION TITLE
[Bug]: Generate unique id for all objects

### DISCUSSION THREAD

**Discussion by shaktippatel** [fabricjs/fabric.js#9445] [Q&A]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

6.0.0-beta14

### In What environments are you experiencing the problem?

Chrome, Node.js

### Node Version (if applicable)

21.0.0

### Link To Reproduction

Next template >= v6

### Steps To Reproduce

**Logic for previous version**

```
function getUniqueId() {
  return getUniqueId.uniqueId++;  
}
getUniqueId.uniqueId = 0;

fabric.Object.prototype.setOptions = (function (setOptions) {
    return function (options) {
        setOptions.apply(this, [options]);
        this.uniqueId = this.uniqueId || getUniqueId();
    };
})(fabric.Object.prototype.setOptions);
```

**New Version**
Not able to set uniqueId with above logic

### Expected Behavior

Generate unique id for all objects in version 6 +

### Actual Behavior

Generate unique id not working due to protected method

### Error Message & Stack Trace

```bash
Generate unique id for all objects in version 6 +
```

**Comment by asturur** [comment]
one way to obtain this is to override the Object.getDefaults method

```ts
  static getDefaults(): Record<string, any> {
    return { uniqueId: myUniQueId(), ...FabricObject.ownDefaults };
  }
```

this is an example and i m sure there are many ways to do it.

**Reply by shaktippatel** [reply]
toObject not returnning **uniqueId** value

**Reply by asturur** [reply]
You specifically mention you want to add it.
Do I have to assume that you were able to add it successfully?

**Reply by gloriousjob** [reply]
The way I would handle this is create subclasses of objects you use (e.g., MyRect from Rect) and  utilize the changes you want. It's more work but it properly separates the original code from your code. You'd have an id generator utility as a separate class.
